### PR TITLE
refactor: Replace old integration tests for blueprint

### PR DESCRIPTION
**Pull request by diegolnasc**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@diegolnasc
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@diegolnasc
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by diegolnasc** [comment]
@bharathkkb Hello,
Locally my tests seem to be working. Can you give me a hint of what is happening in the integration tests?

File: build/int.cloudbuild.yaml [hunk 1]
```diff
@@ -21,143 +21,139 @@ steps:
       - 'TF_VAR_org_id=$_ORG_ID'
       - 'TF_VAR_folder_id=$_FOLDER_ID'
       - 'TF_VAR_billing_account=$_BILLING_ACCOUNT'
-  - id: create-all
+  - id: create all
     waitFor:
       - prepare
     name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
-    args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do create']
+    args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && source_test_env && init_credentials && cd test/integration && RUN_STAGE=init go test -v ./... -p 1 -timeout 0']
```
Comment (bharathkkb) [inline build/int.cloudbuild.yaml:28]: nit: we should use the CLI here and throughout
```suggestion
    args: ['/bin/bash', '-c', 'cft test run all --stage init --verbose']
```

File: build/int.cloudbuild.yaml [hunk 2]
```diff
@@ -21,143 +21,139 @@ steps:
...
-  - id: converge-postgresql-public-local
+  - id: converge mssql-ha-local
     waitFor:
-      - create-all
+      - create all
     name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
-    args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do converge postgresql-public-local']
-    timeout: 3600s
-  - id: verify-postgresql-public-local
+    args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && source_test_env && init_credentials && cd test/integration && RUN_STAGE=apply go test -v ./... -p 1 -timeout 0 -run ^TestMsSqlHaModule']
```
Comment (bharathkkb) [inline build/int.cloudbuild.yaml:34]:
```suggestion
    args: ['/bin/bash', '-c', 'cft test run TestMsSqlHaModule --stage apply --verbose']
```

Comment (bharathkkb) [reply build/int.cloudbuild.yaml:34]: You can use `cft test list` to discover all tests too.

File: build/int.cloudbuild.yaml [hunk 3]
```diff
@@ -21,143 +21,139 @@ steps:
       - 'TF_VAR_org_id=$_ORG_ID'
       - 'TF_VAR_folder_id=$_FOLDER_ID'
       - 'TF_VAR_billing_account=$_BILLING_ACCOUNT'
-  - id: create-all
+  - id: create all
```
Comment (bharathkkb) [inline build/int.cloudbuild.yaml:24]: nit: we prefer to use updated stage names if possible `"init", "apply", "verify", "teardown"`

**Review by bharathkkb — COMMENTED** [review]
This is awesome, thanks for working on this! Some initial feedback below

Tests seem to have failed in `converge mssql-public-local` step. I have seen this error before in https://github.com/terraform-google-modules/terraform-google-sql-db/pull/258#issuecomment-1024433038 so I don't think its related to the refactor.

```
TestMsSqlPublicModule 2022-02-10T04:13:50Z command.go:185: random_id.instance_name_suffix: Creation complete after 2s [id=B9QhYf8]
TestMsSqlPublicModule 2022-02-10T04:13:54Z command.go:185: module.mssql.module.mssql.google_sql_database_instance.default: Creating...
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: ╷
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ Error: Error waiting for Create Instance: error while retrieving operation: googleapi: Error 403: Cloud SQL Admin API has not been used in project PROJECT_NUM before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ Details:
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ [
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   {
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "@type": "type.googleapis.com/google.rpc.Help",
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "links": [
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │       {
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │         "description": "Google developers console API activation",
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │         "url": "https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM"
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │       }
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     ]
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   },
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   {
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "domain": "googleapis.com",
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "metadata": {
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │       "consumer": "projects/PROJECT_NUM",
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │       "service": "sqladmin.googleapis.com"
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     },
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "reason": "SERVICE_DISABLED"
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   }
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ ]
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ , accessNotConfigured
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ 
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   with module.mssql.module.mssql.google_sql_database_instance.default,
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   on ../../../modules/mssql/main.tf line 43, in resource "google_sql_database_instance" "default":
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   43: resource "google_sql_database_instance" "default" {
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ 
TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: ╵
TestMsSqlPublicModule 2022-02-10T04:13:55Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
│ Error: Error waiting for Create Instance: error while retrieving operation: googleapi: Error 403: Cloud SQL Admin API has not been used in project PROJECT_NUM before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.Help",
│     "links": [
│       {
│         "description": "Google developers console API activation",
│         "url": "https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM"
│       }
│     ]
│   },
│   {
│     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
│     "domain": "googleapis.com",
│     "metadata": {
│       "consumer": "projects/PROJECT_NUM",
│       "service": "sqladmin.googleapis.com"
│     },
│     "reason": "SERVICE_DISABLED"
│   }
│ ]
│ , accessNotConfigured
│ 
│   with module.mssql.module.mssql.google_sql_database_instance.default,
│   on ../../../modules/mssql/main.tf line 43, in resource "google_sql_database_instance" "default":
│   43: resource "google_sql_database_instance" "default" {
│ 
╵}
    apply.go:34: 
        	Error Trace:	apply.go:34
        	            				terraform.go:312
        	            				terraform.go:322
        	            				terraform.go:346
        	            				stages.go:31
        	            				terraform.go:346
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; ╷
        	            	│ Error: Error waiting for Create Instance: error while retrieving operation: googleapi: Error 403: Cloud SQL Admin API has not been used in project PROJECT_NUM before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
        	            	│ Details:
        	            	│ [
        	            	│   {
        	            	│     "@type": "type.googleapis.com/google.rpc.Help",
        	            	│     "links": [
        	            	│       {
        	            	│         "description": "Google developers console API activation",
        	            	│         "url": "https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM"
        	            	│       }
        	            	│     ]
        	            	│   },
        	            	│   {
        	            	│     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        	            	│     "domain": "googleapis.com",
        	            	│     "metadata": {
        	            	│       "consumer": "projects/PROJECT_NUM",
        	            	│       "service": "sqladmin.googleapis.com"
        	            	│     },
        	            	│     "reason": "SERVICE_DISABLED"
        	            	│   }
        	            	│ ]
        	            	│ , accessNotConfigured
        	            	│ 
        	            	│   with module.mssql.module.mssql.google_sql_database_instance.default,
        	            	│   on ../../../modules/mssql/main.tf line 43, in resource "google_sql_database_instance" "default":
        	            	│   43: resource "google_sql_database_instance" "default" {
        	            	│ 
        	            	╵}
        	Test:       	TestMsSqlPublicModule
```

**Comment by comment-bot-dev** [comment]
@diegolnasc
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by diegolnasc** [comment]
> This is awesome, thanks for working on this! Some initial feedback below
> 
> Tests seem to have failed in `converge mssql-public-local` step. I have seen this error before in [#258 (comment)](https://github.com/terraform-google-modules/terraform-google-sql-db/pull/258#issuecomment-1024433038) so I don't think its related to the refactor.
> 
> ```
> TestMsSqlPublicModule 2022-02-10T04:13:50Z command.go:185: random_id.instance_name_suffix: Creation complete after 2s [id=B9QhYf8]
> TestMsSqlPublicModule 2022-02-10T04:13:54Z command.go:185: module.mssql.module.mssql.google_sql_database_instance.default: Creating...
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: ╷
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ Error: Error waiting for Create Instance: error while retrieving operation: googleapi: Error 403: Cloud SQL Admin API has not been used in project PROJECT_NUM before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ Details:
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ [
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   {
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "@type": "type.googleapis.com/google.rpc.Help",
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "links": [
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │       {
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │         "description": "Google developers console API activation",
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │         "url": "https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM"
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │       }
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     ]
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   },
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   {
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "domain": "googleapis.com",
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "metadata": {
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │       "consumer": "projects/PROJECT_NUM",
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │       "service": "sqladmin.googleapis.com"
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     },
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │     "reason": "SERVICE_DISABLED"
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   }
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ ]
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ , accessNotConfigured
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ 
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   with module.mssql.module.mssql.google_sql_database_instance.default,
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   on ../../../modules/mssql/main.tf line 43, in resource "google_sql_database_instance" "default":
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │   43: resource "google_sql_database_instance" "default" {
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: │ 
> TestMsSqlPublicModule 2022-02-10T04:13:55Z command.go:185: ╵
> TestMsSqlPublicModule 2022-02-10T04:13:55Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
> │ Error: Error waiting for Create Instance: error while retrieving operation: googleapi: Error 403: Cloud SQL Admin API has not been used in project PROJECT_NUM before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
> │ Details:
> │ [
> │   {
> │     "@type": "type.googleapis.com/google.rpc.Help",
> │     "links": [
> │       {
> │         "description": "Google developers console API activation",
> │         "url": "https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM"
> │       }
> │     ]
> │   },
> │   {
> │     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
> │     "domain": "googleapis.com",
> │     "metadata": {
> │       "consumer": "projects/PROJECT_NUM",
> │       "service": "sqladmin.googleapis.com"
> │     },
> │     "reason": "SERVICE_DISABLED"
> │   }
> │ ]
> │ , accessNotConfigured
> │ 
> │   with module.mssql.module.mssql.google_sql_database_instance.default,
> │   on ../../../modules/mssql/main.tf line 43, in resource "google_sql_database_instance" "default":
> │   43: resource "google_sql_database_instance" "default" {
> │ 
> ╵}
>     apply.go:34: 
>         	Error Trace:	apply.go:34
>         	            				terraform.go:312
>         	            				terraform.go:322
>         	            				terraform.go:346
>         	            				stages.go:31
>         	            				terraform.go:346
>         	Error:      	Received unexpected error:
>         	            	FatalError{Underlying: error while running command: exit status 1; ╷
>         	            	│ Error: Error waiting for Create Instance: error while retrieving operation: googleapi: Error 403: Cloud SQL Admin API has not been used in project PROJECT_NUM before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
>         	            	│ Details:
>         	            	│ [
>         	            	│   {
>         	            	│     "@type": "type.googleapis.com/google.rpc.Help",
>         	            	│     "links": [
>         	            	│       {
>         	            	│         "description": "Google developers console API activation",
>         	            	│         "url": "https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=PROJECT_NUM"
>         	            	│       }
>         	            	│     ]
>         	            	│   },
>         	            	│   {
>         	            	│     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
>         	            	│     "domain": "googleapis.com",
>         	            	│     "metadata": {
>         	            	│       "consumer": "projects/PROJECT_NUM",
>         	            	│       "service": "sqladmin.googleapis.com"
>         	            	│     },
>         	            	│     "reason": "SERVICE_DISABLED"
>         	            	│   }
>         	            	│ ]
>         	            	│ , accessNotConfigured
>         	            	│ 
>         	            	│   with module.mssql.module.mssql.google_sql_database_instance.default,
>         	            	│   on ../../../modules/mssql/main.tf line 43, in resource "google_sql_database_instance" "default":
>         	            	│   43: resource "google_sql_database_instance" "default" {
>         	            	│ 
>         	            	╵}
>         	Test:       	TestMsSqlPublicModule
> ```

That's interesting.
The tests seem to be ok in my environment, at least that explains something.

Anyway I removed the WIP. Can I do something about this?

File: test/integration/mysql-ha/mysql_ha_test.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	mySql.DefineVerify(func(assert *assert.Assertions) {
+		mySql.DefaultVerify(assert)
+
+		instaceNames := []string{mySql.GetStringOutput("name")}
+		op := gcloud.Run(t, fmt.Sprintf("sql instances describe %s --project %s", instaceNames[0], mySql.GetStringOutput("project_id")))
+		assert.Equal(len(op.Get("replicaNames").Array()), 3, "Expected 3 replicas")
+		instaceNames = append(instaceNames, op.Get("replicaNames").Array()[0].String())
+		instaceNames = append(instaceNames, op.Get("replicaNames").Array()[1].String())
+		instaceNames = append(instaceNames, op.Get("replicaNames").Array()[2].String())
+
```
Comment (bharathkkb) [inline test/integration/mysql-ha/mysql_ha_test.go:72]: You can use https://pkg.go.dev/github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test@v0.0.0-20220210045134-f3d60ff938cb/pkg/utils#GetResultStrSlice
```suggestion
		instaceNames = append(instaceNames, utils.GetResultStrSlice(t, op.Get("replicaNames")))

```

**Review by bharathkkb — COMMENTED** [review]
> Can I do something about this?

I think I will do some more digging but this is not a blocker for merging this. Some minor feedback below but overall LGTM

**Comment by comment-bot-dev** [comment]
@diegolnasc
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by diegolnasc** [comment]
> I think I will do some more digging but this is not a blocker for merging this. Some minor feedback below but overall LGTM

I followed most of your suggestions and also hardcoded everything.

**Review by bharathkkb — APPROVED** [review]
Thanks again @diegolnasc

**Pull request merged**